src/coord/src/coord.rs

```diff
@@ -1576,6 +1576,9 @@ impl Coordinator {
                 self.ship_dataflow(df).await;
             }
             self.maybe_begin_caching(source_id, caching_enabled).await;
+            let frontiers =
+                self.new_frontiers(source_id, Some(0), self.logical_compaction_window_ms);
```

Comment:
Before I say anything else -- I've removed this commit from the PR because this got tricky to debug, and I wanted to get the other commits into main / get some momentum going instead of feeling like I'm jumping from one rabbit hole to the next.

I think having `Some(0)` is correct here because this code path is only called when we first create a source, which means that we don't yet have a unmaterialized source to get a since from, and since theres no other inputs, the newly created dataflow can only have a as_of of 0? I do think its legit to ask that this block be placed before we ship the index dataflow, and I can do that.

Reply:
It could also use `df.as_of`